devops/actions/khronos_cts_test/action.yml

```diff
@@ -0,0 +1,123 @@
+    shell: bash
+  - run: apt update && apt install -yqq opencl-headers ocl-icd-opencl-dev
+    shell: bash
+  - name: Checkout LLVM Test Suite
+    uses: ./actions/cached_checkout
+    with:
+      path: khronos_sycl_cts
+      repository: 'KhronosGroup/SYCL-CTS'
+      ref: ${{ inputs.test_ref }}
+      default_branch: 'SYCL-2020'
+      cache_path: "/__w/repo_cache/"
+  - run: |
+      cd khronos_sycl_cts
+      git submodule update --init
+    shell: bash
+  # Remove failed test case
+  - run: |
+      # TODO remove this step when host_task_interop_api.cpp fixed, CMPLRTST-17212
+      rm khronos_sycl_cts/tests/host_task/host_task_interop_api.cpp
+      # PI CUDA kernels only support images with channel types int32, uint32, float, and half.
+      rm khronos_sycl_cts/tests/accessor_legacy/accessor_api_image_core.cpp
+    shell: bash
+  - name: Download compiler toolchain
+    uses: actions/download-artifact@v2
+    with:
+      name: ${{ inputs.sycl_artifact }}
+  - name: Extract SYCL toolchain
+    shell: bash
+    run: |
+      mkdir toolchain
+      tar -xf ${{ inputs.sycl_archive }} -C toolchain
+      rm -f ${{ inputs.sycl_archive }}
+  # TODO these must be in the container
+  - name: Build SYCL-CTS
+    shell: bash
+    run: |
+      echo "::group::CMake configuration"
+      export LD_LIBRARY_PATH=$PWD/toolchain/lib/:$LD_LIBRARY_PATH
+      export PATH=$PWD/toolchain/bin/:$PATH
+      mkdir build
+      cmake  -GNinja -B./build -S./khronos_sycl_cts -DSYCL_IMPLEMENTATION=DPCPP \
+      -DDPCPP_TARGET_TRIPLES=nvptx64-nvidia-cuda \
+      -DCMAKE_CXX_COMPILER="$PWD/toolchain/bin/clang++" \
+      -DSYCL_CTS_EXCLUDE_TEST_CATEGORIES="$PWD/llvm/devops/cts_exclude_filter" \
+      -DSYCL_CTS_ENABLE_OPENCL_INTEROP_TESTS=OFF \
+      -DSYCL_CTS_MEASURE_BUILD_TIMES=ON \
+      -DDPCPP_INSTALL_DIR="$PWD/toolchain" \
+      ${{ inputs.cmake_args }}
+      echo "::endgroup::"
+      echo "::group::Bulid testing"
+      cd build
+      ninja
+      echo "::endgroup::"
+  - name: Run SYCL-CTS
+    shell: bash
+    run: |
+      export LD_LIBRARY_PATH=$PWD/toolchain/lib/:$LD_LIBRARY_PATH
+      export PATH=$PWD/toolchain/bin/:$PATH
+      # TODO make this part of container build
+      export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/rocm/hip/lib/:/opt/rocm/lib
+      export SYCL_DEVICE_FILTER=${{ matrix.sycl_device_filter }}
+      if [ -e /runtimes/oneapi-tbb/env/vars.sh ]; then
+        source /runtimes/oneapi-tbb/env/vars.sh;
+      elif [ -e /opt/runtimes/oneapi-tbb/env/vars.sh ]; then
+        source /opt/runtimes/oneapi-tbb/env/vars.sh;
+      else
+        echo "no TBB vars in /opt/runtimes or /runtimes";
+      fi
+      # TODO remove workaround of FPGA emu bug
+      mkdir -p icd
+      echo /usr/lib/x86_64-linux-gnu/intel-opencl/libigdrcl.so > icd/gpu.icd
+      echo /runtimes/oclcpu/x64/libintelocl.so > icd/cpu.icd
+      echo /opt/runtimes/oclcpu/x64/libintelocl.so > icd/cpu2.icd
+      export OCL_ICD_VENDORS=$PWD/icd
```

Comment:
Why is this needed? I expect the container to be already correctly configured.

Reply:
This is what I have set up to mimic the environment of CUDA LLVM Test Suite, currently CUDA LLVM Test Suite is working well, so I think the environment should be correct. When the test is stable I will do this configuration when building the image.

Reply:
Ok. I'll do review when it's marked as ready for review then.

NOTE: Some of the tests you disabled should be fixed by https://github.com/KhronosGroup/SYCL-CTS/pull/367.
I suggest checking the status using latest SYCL-CTS source and re-enable disabled tests. Ideally we should use the filter file from the Khronos SYCL-CTS repository instead of creating a local duplicate.